competition-simulator/programming.md

```diff
@@ -86,3 +86,53 @@ The `see` method will return a list of visible markers in the arena, but doesn't
 - The following attributes are not available:
   - `Marker.res`
   - `Point.image`
+
+### Time
+
+In the simulated environment, time advances only at the pace that the simulator
+is run. As a result, using `time.time` to know how long your robot has been
+running for or `time.sleep` to wait for some duration will be unreliable.
+
+Instead the simulated `Robot` API provides some alternatives which you are
+encouraged to use instead.
+
+#### Sleeping
+
+If you want to wait for something to happen within the simulation, and you can
+be reasonably confident that it will take a given amount of time, you can use
+`Robot.sleep` method to pause your code for a given duration.
+
+Internally this uses the simulation's own clock and so is suitable for use in
+place of `time.sleep`. Note that, just as with `time.sleep`, while your code is
+sleeping your robot will continue any actions it was doing.
+
+``` python
+R = Robot()
+
+# Blink the output
+R.ruggeduinos[0].digital_write(A_PIN, 1)
+R.sleep()
```

Comment:
Should there be a number as an argument here and should we clarify this is seconds?
```suggestion
R.sleep(1)  # Sleep for 1 second of simulation time
```

Reply:
d'oh, yeah. It would have errored otherwise!

Reply:
67f33b0